sycl/include/sycl/functional.hpp

```diff
@@ -24,13 +24,27 @@ template <typename T = void> using bit_xor = std::bit_xor<T>;
 // std:logical_and/std::logical_or with a non-void type returns bool,
 // sycl requires returning T.
 template <typename T = void> struct logical_and {
-  T operator()(const T &lhs, const T &rhs) const { return lhs && rhs; }
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
I think preview path should just delegate to `std::` versions, similar to lines 18-22.